### PR TITLE
[FLINK-27536][Connectors / Common] Rename method parameter in AsyncSi…

### PR DESCRIPTION
**Pull request by DavidLiu001**
…nkWriter

## What is the purpose of the change

Change the abstract method's parameter naming in AsyncSinkWriter and AsyncSinkWriterTest

From  Consumer<List<RequestEntryT>> requestResult
to  Consumer<List<RequestEntryT>> requestToRetry

This is because the consumer here is supposed to accept a list of requests that need to be retried.

## Brief change log
flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java
flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriterTest.java
Just rename the parameter of requestResult to requestToRetry

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)     no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)    no
  - The serializers: (yes / no / don't know)    no
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)    no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)    no
  - The S3 file system connector: (yes / no / don't know)    no

## Documentation

  - Does this pull request introduce a new feature? (yes / no)    no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java [hunk 1]
```diff
@@ -170,30 +170,30 @@
      * the valid limits of the destination). The logic then needs to create and execute the request
      * asynchronously against the destination (ideally by batching together multiple request entries
      * to increase efficiency). The logic also needs to identify individual request entries that
-     * were not persisted successfully and resubmit them using the {@code requestResult} callback.
+     * were not persisted successfully and resubmit them using the {@code requestToRetry} callback.
```
Comment (dannycranmer) [inline flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java:173]: I understand the Jira said `requestToRetry`, but given this is a `List<>` I think a better name would be `requestsToRetry`, what do you think?

Comment (DavidLiu001) [reply flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java:173]: For example, it is used to call the method of accept() ->  requestToRetry.accept(retryableFailedRecords)
@dannycranmer  If you still think requestsToRetry is better, I will change to  requestsToRetry

Comment (dannycranmer) [reply flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java:173]: ok makes sense

**Comment by dannycranmer** [comment]
@DavidLiu001 you have a [build failure](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=38572&view=logs&j=52b61abe-a3cc-5bde-cc35-1bbe89bb7df5&t=54421a62-0c80-5aad-3319-094ff69180bb) on the CI. Can you please address this and then I will merge. Thanks

**Comment by DavidLiu001** [comment]
Close this pull request and I Will open a new Pull Request

**Comment by dannycranmer** [comment]
@DavidLiu001 in future instead of opening a new PR, you can simply push a new commit to you branch and the PR will automatically update? You can even squash and force push to rewrite the branch completely. Thanks.

**Comment by DavidLiu001** [comment]
Nice,  I will follow your suggestion for the later PR

**Comment by DavidLiu001** [comment]
Nice,  I will follow your suggestion for the later PR

**Comment by dannycranmer** [comment]
Superseded by https://github.com/apache/flink/pull/20360

**Pull request closed without merging**